testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/access/log/ConsoleAccessLogTestCase.java

```diff
@@ -0,0 +1,432 @@
+        queryString.get("key").set("query_string");
+        queryString.get("include-question-mark").set(true);
+
+        executeOperation(client.getControllerClient(), op);
+
+        sendRequest(new BasicNameValuePair("testParam", "testValue"));
+        final Collection<JsonObject> lines = findLines();
+        Assert.assertFalse("Did not find eventSource in " + stdout.toString(), lines.isEmpty());
+        for (JsonObject jsonObject : lines) {
+            Assert.assertEquals("web-access", jsonObject.getString("eventSource"));
+            Assert.assertNull("include-host-name attribute was set to false and should not be included in the output.", jsonObject.get("hostName"));
+            Assert.assertEquals("Expected @version to be 1", "1", jsonObject.getString("@version"));
+            final String timestamp = jsonObject.getString("@timestamp");
+            try {
+                DateTimeFormatter.ofPattern(dateFormat).parse(timestamp);
+            } catch (DateTimeParseException e) {
+                Assert.fail(String.format("Failed to parse date %s with pattern %s: %s", timestamp, dateFormat, e.getMessage()));
+            }
+            Assert.assertEquals("?testParam=testValue", jsonObject.getString("query_string"));
+            Assert.assertEquals("GET", jsonObject.getString("request_method"));
+            Assert.assertEquals(url.getProtocol(), jsonObject.getString("request_scheme"));
+            Assert.assertEquals("/simple-war/simple", jsonObject.getString("request_url"));
+            Assert.assertEquals(HttpStatus.SC_OK, jsonObject.getInt("http_response_code"));
+            Assert.assertEquals(url.getPort(), jsonObject.getInt("port"));
+            Assert.assertTrue(jsonObject.getString("response_header_Content-Type").startsWith("application/json"));
+            Assert.assertNotNull(jsonObject.get("response_header_Content-Encoding"));
+        }
+    }
+
+    private void sendRequest(final NameValuePair... params) throws IOException, URISyntaxException {
+        final URI uri = new URI(url.toString() + "simple");
+        final URIBuilder builder = new URIBuilder(uri);
+        if (params != null && params.length > 0) {
+            builder.setParameters(params);
+        }
+
+        final HttpGet request = new HttpGet(builder.build());
+        try (
+                CloseableHttpClient httpClient = HttpClientBuilder.create().build();
+                CloseableHttpResponse response = httpClient.execute(request)
+        ) {
+            Assert.assertEquals("Failed to access " + uri, HttpStatus.SC_OK, response.getStatusLine().getStatusCode());
+        }
+    }
+
+    private Collection<JsonObject> findLines() throws InterruptedException {
+        // Note this could be a potential spot for a race in the test validation. The console-access-log is
+        // asynchronous so we need to wait to ensure it's fully written to the console.
+        TimeUnit.SECONDS.sleep(TimeoutUtil.adjust(1));
```

Comment:
@jamezp I'm okay with having this, but lets keep an eye to see if we don't have random failures on this test when run on slower machines. If that happens, we will have to switch to an assertion in loop: if fail to find expected lines-> try to find lines again, unless timeout of, say, 30 s or 60 s has passed.

Reply:
+1. I do not generally like things like this, but there seemed to be no other way :)